ui/ducks/confirm-transaction/confirm-transaction.duck.js

```diff
@@ -118,8 +130,32 @@ export default function reducer(state = initState, action = {}) {
         ...state,
         nonce: action.payload,
       };
-    case CLEAR_CONFIRM_TRANSACTION:
-      return initState;
+    case SET_DETAILS_FOR_CONFIRM_TX: {
+      return {
+        ...state,
+        sendTxDetailPerId: {
+          ...state.sendTxDetailPerId,
+          [action.payload.transactionId]: {
+            amountMode: action.payload.amountMode,
+            assetType: action.payload.assetType,
+          },
+        },
+      };
+    }
+    case CLEAR_CONFIRM_TRANSACTION: {
+      // Preserve sendTxDetailPerId
+      return {
+        ...initState,
+        sendTxDetailPerId: state.sendTxDetailPerId,
+      };
+    }
+    case actionConstants.COMPLETED_TX: {
+      const { id } = action.value;
+      // eslint-disable-next-line no-unused-vars
+      const { [id]: _filteredTxDetail, ...newAmountModePerTx } =
+        state.sendTxDetailPerId;
+      return { ...state, sendTxDetailPerId: newAmountModePerTx };
+    }
     default:
```

Comment:
This suggestion may be more readable and does not require the eslint-disable:

```suggestion
    case actionConstants.COMPLETED_TX: {
      const { id } = action.value;
      const sendTxDetailPerId = { ...state.sendTxDetailPerId };
      delete sendTxDetailPerId[id];
      return { ...state, sendTxDetailPerId };
    }
    default:
```